sql/logictest: TestSqlLiteLogic failed

sql/logictest.TestSqlLiteLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4583022&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4583022&tab=artifacts#/) on release-22.1 @ [a1c1879e01ceee79a81693c67a1dba184b5fc1b1](https://github.com/cockroachdb/cockroach/commits/a1c1879e01ceee79a81693c67a1dba184b5fc1b1):


```
=== PAUSE TestSqlLiteLogic/fakedist-disk/slt_good_0.test#03
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_0.test#03
[10:56:51] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_13.test: 5952 statements
[10:56:51] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_11.test: 4501 statements
[10:56:51] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_10.test: 3739 statements
[10:56:51] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_14.test: 7841 statements
[10:56:51] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_0.test: 19 statements
[10:56:51] rng seed: -8916437654246488417

[10:56:52] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_1.test: 887 statements
[10:56:52] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_12.test: 5981 statements
[10:56:53] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_13.test: 6340 statements
[10:56:53] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_11.test: 4988 statements
[10:56:53] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_10.test: 4097 statements
[10:56:53] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_14.test: 8155 statements
[10:56:54] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test: 1003 statements
[10:56:54] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_1.test: 1171 statements
[10:56:54] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_0.test: 26 statements
[10:56:54] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_10.test with config fakedist-disk: 4261 tests, 0 failures
[10:56:54] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_12.test: 6374 statements
[10:56:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_13.test: 6591 statements
[10:56:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_11.test: 5314 statements
[10:56:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_14.test: 8473 statements
[10:56:56] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_1.test: 1444 statements
[10:56:56] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test: 1010 statements
[10:56:56] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_0.test: 33 statements
[10:56:56] --- total progress: 12887145 statements
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_0.test#03
    logic.go:3071: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test:10300: SELECT pk FROM tab3 WHERE col1 > 3356.8 AND col4 IN (SELECT col1 FROM tab3 WHERE (col3 = 6354 AND col3 >= 4129 OR col0 >= 5009 AND (col3 > 964) AND (col4 > 4072.6)) AND col0 BETWEEN 3729 AND 8439 AND col0 >= 4860 OR ((col4 < 2879.73)) OR col1 >= 4543.22 OR col3 < 5765)
        expected success, but found
        (53200) joinreader-mem: memory budget exceeded: 81920 bytes requested, 81920 currently allocated, 102400 bytes in budget
        resource.go:37: in NewBudgetExceededError()
    logic.go:2782: 
         pq: joinreader-mem: memory budget exceeded: 81920 bytes requested, 81920 currently allocated, 102400 bytes in budget
[10:57:46] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test with config fakedist-disk: 2548 tests, 2 failures
    logic.go:3722: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test:10304: error while processing
    logic.go:3722: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test:10304: too many errors encountered, skipping the rest of the input
[10:57:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_1.test: 666 statements
[10:57:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_0.test: 7250 statements
[10:57:47] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_0.test: 360 statements
[10:57:47] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_1.test: 8251 statements
[10:57:47] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_2.test: 958 statements
[10:57:47] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_3.test: 1819 statements
[10:57:47] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/100/slt_good_4.test: 2504 statements
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_0.test#03 (56.80s)
=== RUN   TestSqlLiteLogic/fakedist-disk
    --- FAIL: TestSqlLiteLogic/fakedist-disk (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSqlLiteLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13854

sql/logictest.TestSqlLiteLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4594083&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4594083&tab=artifacts#/) on release-22.1 @ [2d49a7bd0d1ebeff93277233695411d25b2a5d39](https://github.com/cockroachdb/cockroach/commits/2d49a7bd0d1ebeff93277233695411d25b2a5d39):


```
[10:31:56] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_39.test: 5450 statements
[10:31:56] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_14.test: 2810 statements
[10:31:56] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_0.test: 10655 statements
[10:31:57] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_19.test: 9321 statements
[10:31:57] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_15.test: 5003 statements
[10:31:57] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test: 1019 statements
[10:31:57] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_16.test: 3637 statements
[10:31:57] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_17.test: 2755 statements
[10:31:58] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_39.test: 5688 statements
[10:31:58] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_14.test: 3136 statements
[10:31:58] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_0.test: 10819 statements
[10:31:59] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_19.test: 9574 statements
[10:31:59] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_15.test: 5304 statements
[10:31:59] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test: 1060 statements
[10:31:59] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_16.test: 3929 statements
[10:31:59] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_17.test: 3064 statements
[10:32:00] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_39.test: 5935 statements
[10:32:00] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_14.test: 3373 statements
[10:32:00] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_0.test: 11001 statements
[10:32:00] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/1000/slt_good_0.test with config fakedist-disk: 11040 tests, 0 failures
[10:32:01] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_19.test: 9825 statements
[10:32:01] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_15.test: 5722 statements
[10:32:01] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test: 1152 statements
[10:32:01] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_16.test: 4267 statements
[10:32:02] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_17.test: 3336 statements
[10:32:02] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_39.test: 6115 statements
[10:32:02] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_19.test with config fakedist-disk: 10052 tests, 0 failures
[10:32:02] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_14.test: 3687 statements
[10:32:02] --- total progress: 12606785 statements
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_0.test#03
    logic.go:3071: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test:10300: SELECT pk FROM tab3 WHERE col1 > 3356.8 AND col4 IN (SELECT col1 FROM tab3 WHERE (col3 = 6354 AND col3 >= 4129 OR col0 >= 5009 AND (col3 > 964) AND (col4 > 4072.6)) AND col0 BETWEEN 3729 AND 8439 AND col0 >= 4860 OR ((col4 < 2879.73)) OR col1 >= 4543.22 OR col3 < 5765)
        expected success, but found
        (53200) joinreader-mem: memory budget exceeded: 61440 bytes requested, 51200 currently allocated, 102400 bytes in budget
        resource.go:37: in NewBudgetExceededError()
    logic.go:2782: 
         pq: joinreader-mem: memory budget exceeded: 61440 bytes requested, 51200 currently allocated, 102400 bytes in budget
[10:32:45] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test with config fakedist-disk: 2548 tests, 2 failures
    logic.go:3722: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test:10304: error while processing
    logic.go:3722: /go/src/github.com/cockroachdb/sqllogictest/test/index/between/1000/slt_good_0.test:10304: too many errors encountered, skipping the rest of the input
[10:32:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_16.test: 9973 statements
[10:32:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_18.test: 4742 statements
[10:32:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_4.test: 1071 statements
[10:32:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby/10/slt_good_14.test: 10045 statements
[10:32:46] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_16.test: 19 statements
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_0.test#03 (57.03s)
=== RUN   TestSqlLiteLogic/fakedist-disk
    --- FAIL: TestSqlLiteLogic/fakedist-disk (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSqlLiteLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestSqlLiteLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4614377&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4614377&tab=artifacts#/) on release-22.1 @ [f5b42a4e3bbdadb25aee89a41871239a624d538d](https://github.com/cockroachdb/cockroach/commits/f5b42a4e3bbdadb25aee89a41871239a624d538d):


```
[10:40:13] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 6854 statements
[10:40:13] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 7270 statements
[10:40:14] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 1438 statements
[10:40:14] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 6980 statements
[10:40:14] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 7569 statements
[10:40:14] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 4528 statements
[10:40:14] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 7294 statements
[10:40:15] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 6863 statements
[10:40:15] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 6951 statements
[10:40:15] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 7281 statements
[10:40:16] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 1539 statements
[10:40:16] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 6991 statements
[10:40:16] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 7579 statements
[10:40:16] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 4540 statements
[10:40:16] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 7303 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 1731 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 6961 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 6875 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 7301 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 7010 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 7586 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 4627 statements
[10:40:18] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 7310 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 1903 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 6972 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 7020 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 6959 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 7313 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 7599 statements
[10:40:20] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 7393 statements
[10:40:21] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 4637 statements
    logic.go:3071: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test:8161: SELECT pk FROM tab2 WHERE 8975 <= col3 AND col0 IN (SELECT col3 FROM tab2 WHERE (((3603.65 >= col4))) OR 631 < col0)
        expected success, but found
        (53200) joinreader-mem: memory budget exceeded: 81920 bytes requested, 81920 currently allocated, 102400 bytes in budget
        resource.go:37: in NewBudgetExceededError()
    logic.go:2782: 
         pq: joinreader-mem: memory budget exceeded: 81920 bytes requested, 81920 currently allocated, 102400 bytes in budget
[10:40:21] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test with config fakedist-disk: 2077 tests, 2 failures
    logic.go:3722: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test:8165: error while processing
    logic.go:3722: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test:8165: too many errors encountered, skipping the rest of the input
[10:40:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 7141 statements
[10:40:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 7057 statements
[10:40:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 6970 statements
[10:40:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 7321 statements
[10:40:23] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 7605 statements
[10:40:23] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 7401 statements
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_0.test#06 (20.87s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSqlLiteLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestSqlLiteLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4619914&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4619914&tab=artifacts#/) on release-22.1 @ [a25a145510d5d0bc7570db327a21cd3fcde21980](https://github.com/cockroachdb/cockroach/commits/a25a145510d5d0bc7570db327a21cd3fcde21980):


```
--- FAIL: TestSqlLiteLogic (23609.40s)
=== RUN   TestSqlLiteLogic/fakedist-disk/slt_good_0.test#06
=== PAUSE TestSqlLiteLogic/fakedist-disk/slt_good_0.test#06
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_0.test#06
[10:36:30] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 9227 statements
[10:36:30] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_2.test: 6748 statements
[10:36:30] rng seed: -5013031339472008183

[10:36:31] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 10931 statements
[10:36:31] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_3.test: 8189 statements
[10:36:31] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_3.test: 9796 statements
[10:36:32] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_1.test: 5807 statements
[10:36:32] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 1667 statements
[10:36:32] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 9398 statements
[10:36:33] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test: 11003 statements
[10:36:33] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 1005 statements
[10:36:33] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_2.test: 6758 statements
[10:36:33] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_3.test: 8200 statements
[10:36:33] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/in/1000/slt_good_1.test with config fakedist-disk: 11024 tests, 0 failures
[10:36:33] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_3.test: 9992 statements
[10:36:34] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_1.test: 5941 statements
[10:36:34] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 9558 statements
[10:36:34] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 1675 statements
[10:36:35] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_2.test: 6809 statements
[10:36:35] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test: 1011 statements
[10:36:35] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_3.test: 8207 statements
[10:36:35] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_3.test: 10090 statements
[10:36:36] --- total progress: 12997540 statements
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_0.test#06
    logic.go:3071: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test:8156: SELECT pk FROM tab2 WHERE col3 >= 8975 AND col0 IN (SELECT col3 FROM tab2 WHERE (((col4 <= 3603.65))) OR col0 > 631)
        expected success, but found
        (53200) joinreader-mem: memory budget exceeded: 81920 bytes requested, 81920 currently allocated, 102400 bytes in budget
        resource.go:37: in NewBudgetExceededError()
    logic.go:2782: 
         pq: joinreader-mem: memory budget exceeded: 81920 bytes requested, 81920 currently allocated, 102400 bytes in budget
[10:36:54] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test with config fakedist-disk: 2076 tests, 2 failures
    logic.go:3722: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test:8160: error while processing
    logic.go:3722: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_0.test:8160: too many errors encountered, skipping the rest of the input
[10:36:54] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_32.test: 19 statements
[10:36:54] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_3.test: 8840 statements
[10:36:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_1.test: 6420 statements
[10:36:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 1937 statements
[10:36:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/100/slt_good_2.test: 7403 statements
[10:36:55] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/10/slt_good_30.test: 1901 statements
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_0.test#06 (25.81s)
=== RUN   TestSqlLiteLogic/fakedist-disk
    --- FAIL: TestSqlLiteLogic/fakedist-disk (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSqlLiteLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/logictest.TestSqlLiteLogic [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4645228&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4645228&tab=artifacts#/) on release-22.1 @ [26c05e15d3752f47a91592b5b870360c30c57dd5](https://github.com/cockroachdb/cockroach/commits/26c05e15d3752f47a91592b5b870360c30c57dd5):


```
[10:07:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 5866 statements
[10:07:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 8723 statements
[10:07:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2053 statements
[10:07:22] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 3093 statements
[10:07:23] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 9126 statements
[10:07:24] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 5246 statements
[10:07:24] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 9151 statements
[10:07:24] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 9430 statements
[10:07:24] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 5878 statements
[10:07:24] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2259 statements
[10:07:25] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 8735 statements
[10:07:25] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 9135 statements
[10:07:25] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 3183 statements
[10:07:26] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 5295 statements
[10:07:26] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 9523 statements
[10:07:26] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 5885 statements
[10:07:26] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2329 statements
[10:07:26] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test: 9161 statements
[10:07:27] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 8756 statements
[10:07:27] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 9152 statements
[10:07:27] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 3192 statements
[10:07:27] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_2.test with config fakedist-disk: 9390 tests, 0 failures
[10:07:28] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 5306 statements
[10:07:28] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2518 statements
[10:07:28] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_5.test: 9534 statements
[10:07:28] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 5906 statements
[10:07:29] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_4.test: 8765 statements
[10:07:29] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_3.test: 9175 statements
[10:07:29] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 3199 statements
[10:07:30] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_0.test: 5337 statements
[10:07:30] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test: 2613 statements
[10:07:30] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 5912 statements
[10:07:30] --- total progress: 11638702 statements
=== CONT  TestSqlLiteLogic/fakedist-disk/slt_good_1.test#04
    logic.go:3071: 
        
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test:48876: SELECT pk FROM tab4 WHERE col0 IN (SELECT col3 FROM tab4 WHERE (col3 <= 5868))
        expected success, but found
        (53200) joinreader-mem: memory budget exceeded: 71680 bytes requested, 81920 currently allocated, 102400 bytes in budget
        resource.go:37: in NewBudgetExceededError()
    logic.go:2782: 
         pq: joinreader-mem: memory budget exceeded: 71680 bytes requested, 81920 currently allocated, 102400 bytes in budget
[10:09:23] --- done: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test with config fakedist-disk: 10410 tests, 2 failures
    logic.go:3722: 
        /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test:48880: error while processing
    logic.go:3722: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/1000/slt_good_1.test:48880: too many errors encountered, skipping the rest of the input
[10:09:24] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/commute/100/slt_good_5.test: 117 statements
[10:09:25] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/10/slt_good_1.test: 7439 statements
[10:09:25] --- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/delete/1/slt_good_0.test: 4305 statements
        --- FAIL: TestSqlLiteLogic/fakedist-disk/slt_good_1.test#04 (142.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSqlLiteLogic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/rowexec/joinreader.go
+++ b/pkg/sql/rowexec/joinreader.go
@@ -420,12 +420,16 @@ func newJoinReader(
 		}
 	}
 
-	// We will create a memory monitor with at least 100KiB of memory limit
-	// since the join reader doesn't know how to spill its in-memory state to
-	// disk (separate from the buffered rows). It is most likely that if the
-	// target limit is below 100KiB, then we're in a test scenario and we don't
-	// want to error out.
-	const minMemoryLimit = 100 << 10
+	// We will create a memory monitor with a hard memory limit since the join
+	// reader doesn't know how to spill its in-memory state to disk (separate
+	// from the buffered rows). It is most likely that if the target limit is
+	// really low then we're in a test scenario and we don't want to error out.
+	minMemoryLimit := int64(8 << 20)
+	// Streamer can handle lower memory limit and doing so makes testing at
+	// the limits more efficient.
+	if jr.usesStreamer {
+		minMemoryLimit = 100 << 10
+	}
 	memoryLimit := execinfra.GetWorkMemLimit(flowCtx)
 	if memoryLimit < minMemoryLimit {
 		memoryLimit = minMemoryLimit

